Frameworks/CoreGraphics/CGContext.mm

```diff
@@ -17,296 +17,526 @@
 #import <StubReturn.h>
 #import <Starboard.h>
 #import <math.h>
+
+#import "CGSurfaceInfoInternal.h" // TODO(DH) Evaluate the need for this header.
+
 #import <CoreGraphics/CGContext.h>
 #import <CoreGraphics/CGPath.h>
 #import <CoreGraphics/CGLayer.h>
-#import "CGColorSpaceInternal.h"
-#import "CGContextInternal.h"
-#include "LoggingNative.h"
-#import "_CGLifetimeBridgingType.h"
 #import <CoreGraphics/CGAffineTransform.h>
 #import <CoreGraphics/CGGradient.h>
-#import <Foundation/NSString.h>
-#import <UIKit/UIImage.h>
-#import <UIKit/UIFont.h>
-#import <UIKit/UIColor.h>
-#import "CGSurfaceInfoInternal.h"
-#import <pthread.h>
+#import "CGColorSpaceInternal.h"
+#import "CGContextInternal.h"
 
-static const wchar_t* TAG = L"CGContext";
+#import <CFRuntime.h>
+
+#include <COMIncludes.h>
+#import <d2d1.h>
+#import <d2d1_1.h>
+#import <d2d1effects_2.h>
+#import <wrl/client.h>
+#include <COMIncludes_end.h>
+#import <LoggingNative.h>
+
+#import <list>
+#import <vector>
+#import <stack>
+
+using namespace Microsoft::WRL;
+
+static inline D2D_RECT_F __CGRectToD2D(CGRect rect) {
+    return {
+        rect.origin.x, rect.origin.y, rect.origin.x + rect.size.width, rect.origin.y + rect.size.height,
+    };
+}
+
+struct __CGContextDrawingState {
+    // This is populated when the state is saved, and contains the D2D parameters that CG does not know.
+    ComPtr<ID2D1DrawingStateBlock> d2dState;
+
+    // Fills
+    ComPtr<ID2D1Brush> fillBrush;
+
+    // Strokes
+    ComPtr<ID2D1Brush> strokeBrush;
+    ComPtr<ID2D1StrokeStyle> strokeStyle;
+    D2D1_STROKE_STYLE_PROPERTIES strokeProperties{
+        D2D1_CAP_STYLE_FLAT,
+        D2D1_CAP_STYLE_FLAT,
+        D2D1_CAP_STYLE_FLAT,
+        D2D1_LINE_JOIN_MITER,
+        10.f, // Default from Reference Docs
+        D2D1_DASH_STYLE_SOLID,
+        0.f,
+    };
+    std::vector<CGFloat> dashes;
+
+    CGFloat lineWidth = 1.0f;
+
+    CGFloat flatness;
+
+    // Clipping
+    CGMutablePathRef currentClippingPath;
+
+    // Image Drawing
+    D2D1_INTERPOLATION_MODE bitmapInterpolationMode;
+
+    // Userspace Coordinate Transformation
+    CGAffineTransform transform = CGAffineTransformIdentity;
+    CGAffineTransform textMatrix = CGAffineTransformIdentity;
+
+    // Alpha Blending
+    CGFloat alpha = 1.0f;
+};
+
+struct __CGContext {
+    struct __CGContextImpl {
+        ComPtr<ID2D1RenderTarget> renderTarget;
+
+        // Calculated at creation time, this transform flips CG's drawing commands,
+        // anchored in the bottom left, to D2D's top-left coordinate system.
+        CGAffineTransform cgCompatibilityTransform;
+
+        std::stack<__CGContextDrawingState> stateStack;
+
+        woc::unique_cf<CGMutablePathRef> currentPath;
+
+        bool allowsAntialiasing;
```

Comment:
Good point. I crosschecked, as I thought CF allocated with calloc. It doesn't :)